onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -637,22 +660,22 @@ void TreeEnsembleCommon<InputType, ThresholdType, OutputType>::ComputeAgg(concur
   }
 }  // namespace detail
 
-#define TREE_FIND_VALUE(CMP)                                    \
-  if (has_missing_tracks_) {                                    \
-    while (root->is_not_leaf()) {                               \
-      val = x_data[root->feature_id];                           \
-      root += (val CMP root->value_or_unique_weight ||          \
-               (root->is_missing_track_true() && _isnan_(val))) \
-                  ? root->truenode_inc_or_first_weight          \
-                  : root->falsenode_inc_or_n_weights;           \
-    }                                                           \
-  } else {                                                      \
-    while (root->is_not_leaf()) {                               \
-      val = x_data[root->feature_id];                           \
-      root += val CMP root->value_or_unique_weight              \
-                  ? root->truenode_inc_or_first_weight          \
-                  : root->falsenode_inc_or_n_weights;           \
-    }                                                           \
+#define TREE_FIND_VALUE(CMP)                                   \
+  if (has_missing_tracks_) {                                   \
+    while (root->is_not_leaf()) {                              \
+      val = x_data[root->feature_id];                          \
+      root = (val CMP root->value_or_unique_weight ||          \
+              (root->is_missing_track_true() && _isnan_(val))) \
+                 ? root->truenode_or_weight.ptr                \
+                 : root + 1;                                   \
```

Comment:
root++ or maybe rewrite it with if... I don't know if the compiler detects that root + 1 is equivalent to increment root.

Reply:
I would actually prefer it written as is. As for what the compiler does I think they generate virtually the same instructions: https://godbolt.org/z/9eGha871s.